python/cudf/cudf/core/column/categorical.py

```diff
@@ -321,9 +321,9 @@ def children(self):
             codes_column = column.build_column(
                 data=codes_column.base_data,
                 dtype=codes_column.dtype,
-                mask=codes_column.base_mask,
+                mask=self.base_mask,
                 size=self.size,
-                offset=self.offset + codes_column.offset,
+                offset=self.offset,
             )
```

Comment:
The problem with this is say I do `cupy.asarray(col.children[0])` it will be incorrect since `col.children[0]` will have an offset. We should do the proper pointer arithmetic and copy construct a new mask to zero the offset as needed.